### PR TITLE
feat: update button to CSS-based implementation

### PR DESCRIPTION
**Pull request by toshusai**
## やったこと

- Changed the Button component to be CSS-based.
- Removed inheritance from Clickable and added the ability to inject any component using the `as` property.

## チェックリスト

不要なチェック項目は消して構いません

- [ ] 破壊的変更がある場合には、対象のパッケージのメジャーバージョンが上がっていることを確認した
- [ ] 追加したコンポーネントが index.ts から再 export されている
- [ ] README やドキュメントに影響があることを確認した

### CODE REVIEW

File: packages/react/package.json [hunk 1]
```diff
@@ -5,8 +5,14 @@
   "main": "./dist/index.cjs.js",
   "module": "./dist/index.esm.js",
   "exports": {
-    "require": "./dist/index.cjs.js",
-    "default": "./dist/index.esm.js"
+    ".": {
+      "require": "./dist/index.cjs.js",
+      "default": "./dist/index.esm.js"
+    },
+    "./dist/index.css": {
```
Comment (naporin0624) [inline packages/react/package.json:12]: How about changing `./dist/index.css` to `./index.css`?

**Review by naporin0624 — COMMENTED** [review]
It looks like a very good change! I've just made a few comments on parts that concern me.

File: packages/react/src/components/Button/__snapshots__/index.story.storyshot
Comment (naporin0624) [inline packages/react/src/components/Button/__snapshots__/index.story.storyshot:1]: Now that the styling has been consolidated into a single CSS file, we can remove the styling from the snapshot!
After completing the migration to v4, I'd like to remove jest-styled-components.

File: packages/react/package.json
[the remark this answers is not in the document]
Comment (toshusai) [reply packages/react/package.json:1]: Thank you for the suggestion. Let's try it out in a separate PR.

File: packages/react/package.json (hunk 1 above)
Comment (toshusai) [reply packages/react/package.json:12]: That's a great idea. Let's consider it before the release of v4.

File: packages/react/src/components/Button/index.css
[the remark this answers is not in the document]
Comment (toshusai) [reply packages/react/src/components/Button/index.css:1]: Thanks for the suggestion. I've added it.

**Review by naporin0624 — APPROVED** [review]

**Review by mimokmt — APPROVED** [review]

**Pull request merged**